.github/workflows/weekly.yml

```diff
@@ -27,25 +27,13 @@ jobs:
     strategy:
       matrix:
         flink_branches: [{
-          flink: 1.16-SNAPSHOT,
-          branch: main
-        }, {
-          flink: 1.17-SNAPSHOT,
-          branch: main
-        }, {
           flink: 1.18-SNAPSHOT,
           jdk: '8, 11, 17',
           branch: main
         }, {
           flink: 1.19-SNAPSHOT,
           jdk: '8, 11, 17, 21',
           branch: main
-        }, {
-          flink: 1.16.2,
-          branch: v3.1
-        }, {
-          flink: 1.17.1,
-          branch: v3.1
         }, {
```

Comment:
should we remove ci for v3.1 ?
I guess this change should not be merged to v3.1 and as a result v3.1 should still continue working/being able to build with 1.16, 1.17
or did I miss anything?

Reply:
if we are bumping to 4.0.0, I agree.
but any incompatible change between flink versions (not need break change, could be a new method we need to use as in this case) will cause that we made a major release??

Reply:
IIRC idealy we should support currently supported Flink versions (1.17, 1.18, 1.19 and 1.16 (probably will be not supported after 1.19 release)).
I don't know another way to have all of them supported with out branching (4.0.0 or 3.2.0 doesn't really matter)

Reply:
We could release 3.2 (3.1.X) with current main code giving support for 1.17 and 1.18 (1.18 still not support by any release), I don't know why still don't have release an 1.18 compatible version, any blocker?

And them merge this and release again with (minor or major)..